SmartDeviceLink/NSMutableDictionary+Store.m

```diff
@@ -33,30 +33,23 @@ - (nullable id)sdl_objectForName:(SDLName)name ofClass:(Class)classType {
     }
 }
 
-- (NSMutableArray *)sdl_objectsForName:(SDLName)name ofClass:(Class)classType {
-    NSMutableArray *array = [self sdl_objectForName:name];
-    if ([array count] < 1 || [[array objectAtIndex:0] isKindOfClass:classType.class]) {
+- (NSArray *)sdl_objectsForName:(SDLName)name ofClass:(Class)classType {
+    NSArray *array = [self sdl_objectForName:name];
+    if ([array count] < 1 || [array.firstObject isMemberOfClass:classType.class]) {
+        // It's an array of the actual class type, just return
         return array;
     } else {
+        // It's an array of dictionaries, make them into their class type
         NSMutableArray *newList = [NSMutableArray arrayWithCapacity:[array count]];
         for (NSDictionary<NSString *, id> *dict in array) {
             [newList addObject:[[classType alloc] initWithDictionary:dict]];
         }
-        return newList;
+        return [NSArray arrayWithArray:newList];
     }
 }
 
-- (NSMutableArray<SDLEnum> *)sdl_enumsForName:(SDLName)name {
-    NSMutableArray<SDLEnum> *array = [self sdl_objectForName:name];
-    if ([array count] < 1) {
-        return array;
-    } else {
-        NSMutableArray<SDLEnum> *newList = [NSMutableArray arrayWithCapacity:[array count]];
-        for (SDLEnum enumString in array) {
-            [newList addObject:enumString];
-        }
-        return newList;
-    }
+- (NSArray<SDLEnum> *)sdl_enumsForName:(SDLName)name {
```

Comment:
If `sdl_enumsForName:` just does `sdl_objectForName:`, should we remove that function and just use `sdl_objectForName:`?